intelmq/bots/collectors/http/collector_http_stream.py

```diff
@@ -8,12 +22,30 @@
 
 class HTTPStreamCollectorBot(CollectorBot):
 
+    def init(self):
+        if getattr(self.parameters, 'url', False) and \
+           not getattr(self.parameters, 'http_url', False):
+            self.logger.warning("Parameter 'url' is deprecated, use 'http_url' instead.")
+            self.parameters.http_url = self.parameters.url
+        self.set_request_parameters()
+
     def process(self):
+        self.logger.info("Connecting to stream at %r." %
+                         self.parameters.http_url)
+
         try:
-            req = requests.get(self.parameters.url, stream=True)
+            req = requests.get(url=self.parameters.http_url, auth=self.auth,
+                               proxies=self.proxy, headers=self.http_header,
+                               verify=self.http_verify_cert,
+                               cert=self.ssl_client_cert, stream=True,
+                               timeout=self.http_timeout)
         except requests.exceptions.ConnectionError:
-            raise ValueError('Connection Failed.')
+            self.logger.exception('Connection Failed.')
         else:
+            if req.status_code // 100 != 2:
+                raise ValueError('HTTP response status code was {}.'
+                                 ''.format(req.status_code))
+
             for line in req.iter_lines():
                 if self.parameters.strip_lines:
                     line = line.strip()
```

Comment:
As opposed to the Pedros collector, the HTTP Stream collector uses requests and iterates over all lines using requests' methods and sends lines as individual reports. So your proposal does not apply.

Also, we are using this bot in production and haven't encountered any problems so far.